### PR TITLE
Add option to execute Tests for release

### PR DESCRIPTION
**Pull request by hdm23061993**
## Fixes Issue
#115

## Changes proposed

- Add a button to 'Run test' on the chart revision page
- Create a new POST API to execute the `helm test`
- Show tests response in a new modal

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [X] My change requires changes to the documentation.
- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots

image

image

### CODE REVIEW

**Comment by undera** [comment]
If PR is not complete, consider turning it into "draft". It's a handy feature in GH, exactly for that.

**Comment by undera** [comment]
Note that this will need to be ported to v1.0

**Comment by hdm23061993** [comment]
@undera  Please help review

Will port to be compatible with release v1.0.0 once this is merged

File: pkg/dashboard/handlers/helmHandlers.go [hunk 1]
```diff
@@ -176,6 +176,22 @@ func (h *HelmHandler) Install(c *gin.Context) {
...
+		return
+	}
+
+	out, err := h.Data.RunTests(qp.Namespace, qp.Name)
+	if err != nil {
+		_ = c.AbortWithError(http.StatusInternalServerError, err)
+		return
+	}
+
+	c.String(http.StatusAccepted, out)
```
Comment (undera) [inline pkg/dashboard/handlers/helmHandlers.go:192]: The tests can actually have output and logs. Shouldn't we block until those are returned, and show the result in UI?

Comment (undera) [reply pkg/dashboard/handlers/helmHandlers.go:192]: I mean `StatusAccepted`  has meaning "the request is accepted for processing, but it is not complete. If it is complete, the status code should be 200 OK

File: pkg/dashboard/subproc/data.go [hunk 2]
```diff
@@ -278,6 +278,17 @@ func (d *DataLayer) ChartInstall(namespace string, name string, repoChart string
 	return out, nil
 }
 
+func (d *DataLayer) RunTests(namespace string, name string) (string, error) {
+	cmd := []string{"test", name, "--namespace", namespace}
```
Comment (undera) [inline pkg/dashboard/subproc/data.go:282]: add `--logs` here?

Comment (hdm23061993) [reply pkg/dashboard/subproc/data.go]: Handled in new commit

File: pkg/dashboard/handlers/helmHandlers.go (hunk 1 above)
Comment (hdm23061993) [reply pkg/dashboard/handlers/helmHandlers.go]: Handled in new commit

**Pull request merged**